### PR TITLE
Chore: Add CODEOWNERS

### PR DESCRIPTION
**Pull request by qcasey**
## Proposed change

Creates CODEOWNERS file, as [documented here](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners).

> Code owners are automatically requested for review when someone opens a pull request that modifies code that they own.

This PR also removes the explicit request for team tagging from the PR template. The CODEOWNERS should do it automatically.

Not covered here, but if we wanted we can **require** that reviews come from the CODEOWNERS teams as part of Branch Protection. I don't see a downside to this, since we require review anyway? It would just make that requirement a little more strict as the reviewer(s) would need to be one of the CODEOWNERS. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Review by shamoon — APPROVED** [review]
Perfect thanks! I was looking at this exact thing the other day!

Ps. Should we just merge this to `main`?

**Comment by qcasey** [comment]
> Ps. Should we just merge this to main?

It can be in main, but the CODEOWNERS operates on the base branch of the PR, so it should definitely be in `dev`

> For code owners to receive review requests, the CODEOWNERS file must be on the base branch of the pull request. For example, if you assign @octocat as the code owner for .js files on the gh-pages branch of your repository, @octocat will receive review requests when a pull request with changes to .js files is opened between the head branch and gh-pages.

**Comment by tribut** [comment]
Does this mean we effectively require three reviews for some PRs now?

**Comment by qcasey** [comment]
> Does this mean we effectively require three reviews for some PRs now?

I haven't made the change to Branch Protection yet, but no I don't think that'd be the case. I'll ask in Matrix about making the change.

For `dev` we currently require 1 approving review from anyone with write access to the org. The change would require 1 approving review from someone deemed a code owner.

The `non-trivial` requirements are more an honor-system and can't be enforced by github.

**Comment by shamoon** [comment]
Yea I think you should go ahead and change the rules according to this to essentially keep as-is, I think has been working well, yea?

**Pull request merged**
